File: bk2ch17p704takeAPicture2/ch30p962takeAPicture2/SecondViewController.m


#import "SecondViewController.h"
#import "ViewController.h"

@interface SecondViewController ()

@property (nonatomic, strong) UIImage* image;
@property (nonatomic, weak) IBOutlet UIImageView* iv;


@end

@implementation SecondViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil image:(UIImage*)im
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        self.title = @"Decide";
        self.navigationItem.rightBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:@"Use" style:UIBarButtonItemStylePlain target:self action:@selector(doUse:)];
        self->_image = im;
    }
    return self;
}

- (void) viewWillAppear:(BOOL)animated {
    [super viewWillAppear: animated];
    [self.navigationController setNavigationBarHidden:NO animated:NO];
    self.iv.image = self.image;
}

- (void) viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
    [self.navigationController setNavigationBarHidden:YES animated:YES];
}

- (void) doUse: (id) sender {
    ViewController* vc = (id)self.presentingViewController;
    [vc doUse: self.image];
}


@end

File: iOS8bookExamples/bk2ch17p704takeAPicture2/ch30p962takeAPicture2/SecondViewController.swift


import UIKit

class SecondViewController: UIViewController {

    var image : UIImage!
    @IBOutlet var iv : UIImageView!
    
    init(image im:UIImage!) {
        self.image = im
        super.init(nibName: "SecondViewController", bundle: nil)
        self.title = "Decide"
        self.navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Use", style: .Plain, target: self, action: "doUse:")
    }
    
    required init(coder: NSCoder) {
        fatalError("NSCoding not supported")
    }
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        self.navigationController?.setNavigationBarHidden(false, animated: false)
        self.iv.image = self.image
    }
    
    override func viewWillDisappear(animated: Bool) {
        super.viewWillDisappear(animated)
        self.navigationController?.setNavigationBarHidden(true, animated: true)
    }
    
    func doUse(sender:AnyObject) {
        let vc = self.presentingViewController as! ViewController
        vc.doUse(self.image)
    }
}
